chain-connect/src/GalachainConnectClient.ts

```diff
@@ -26,12 +26,16 @@ export class GalachainConnectClient {
   #provider: BrowserProvider | undefined;
   #chainCodeUrl: string;
 
-  get address() {
+  get ethAddress() {
     return `eth|${this.#address}`;
   }
 
+  get galachainAddress() {
+    return getAddress(this.#address);
+  }
+
   set address(val: string) {
-    this.#address = val.replace(/0x|eth\|/, "");
+    this.#address = `0x${val.replace(/0x|eth\|/, "")}`;
```

Comment:
To save the initial correct checksum address, this should be:
`this.#address = getAddress('0x${val.replace(/0x|eth\|/, "")}')`